docs/docs/developer-guides/cw-substrate-cli/walkthrough.md

```diff
@@ -0,0 +1,142 @@
+make use of SS58 representations in the contracts as well, but for now, this is how we do it.
+
+So the command will be:
+```sh
+cw-substrate-cli substrate -c ws://127.0.0.1:9988 -n alice \
+    tx instantiate \
+    -c 1 \
+    -s randomsalt \
+    -l our-fancy-cw20base-contract \
+    -g 10000000000 \
+    -m '{"name":"Picasso","symbol":"PICA","decimals":6,"initial_balances":[{"amount":"1000000000000","address":"0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48"}]}'
+```
+
+See that apart from the address of the contract that is instantiated, we also get the raw CosmWasm events.
+
+Output:
+```
+[ + ] Contract instantiated.
+    - Contract address: 5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX
+- Event: instantiate
+    - Contract: 5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX
+    - Attributes:
+        - _contract_address: 0x7c888b21f31f0cec5149830bb0f8f20b949e2b589c240d223625f53df694ca49
+        - code_id: 1
+```
+
+## Execute a transfer
+
+Let's transfer some amount from `Bob` to `Charlie`.
+
+Note that as the signer, we need to use `bob` instead of `alice` because the signer here will be both the `signer` of the extrinsic, and the `sender`
+of the contract message. Since we want to transfer from `bob`, `bob` needs to be the caller of the `execute` call.
+
+Also, the contract address used here is `5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX` which should be the same for you as well since the algorithm for the address
+generation that we use is based on the instantiate parameters that we provide, not on some random values or chain state. But if for some reason, you get a different
+address, use that address to execute the contract.
+
+```sh
+cw-substrate-cli substrate -c ws://127.0.0.1:9988 -n bob \
+    tx execute \
+    -c  5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX \
+    -g 10000000000 \
+    -m '{"transfer":{"amount":"1000","recipient":"0xd64439add16b49b6b68ac74e1b28a73a8491501ab7e0e829716f580947a4bd7e"}}'
+```
+
+Output:
+```
+[ + ] Contract executed.
+- Event: execute
+    - Contract: 5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX
+    - Attributes:
+          - _contract_address: 0x7c888b21f31f0cec5149830bb0f8f20b949e2b589c240d223625f53df694ca49
+- Event: wasm
+    - Contract: 5EszMeNDPmy4orcLEHRLiJawAt5xAvfK5VH7REV8bpB1jtjX
+    - Attributes:
+        - action: transfer
+        - from: 0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48
+        - to: 0xd64439add16b49b6b68ac74e1b28a73a8491501ab7e0e829716f580947a4bd7e
+        - amount: 1000
+        - _contract_address: 0x7c888b21f31f0cec5149830bb0f8f20b949e2b589c240d223625f53df694ca49
+```
+  
+  
+## Query the balance
+
+Although you can that the events clearly shows the transfer happened. Let's query the contract to check out our balance to make sure. Since the query is not a transaction
```

Comment:
```suggestion
Although you can see that the events clearly show the transfer happened. Let's query the contract to check out our balance to make sure. Since the query is not a transaction
```
Nit